src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -87428,5 +87428,114 @@ static void M(string? x)
             var comp = CreateCompilation(new[] { source, EnsuresNotNullAttributeDefinition }, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        [WorkItem(34841, "https://github.com/dotnet/roslyn/issues/34841")]
+        public void PartialClassWithConstraints_01()
+        {
+            var source1 =
+@"#nullable enable
+using System;
+partial class C<T> where T : IEquatable<T>
+{
+}";
+            var source2 =
+@"using System;
+partial class C<T> where T : IEquatable<T>
+{
+}";
+            var comp = CreateCompilation(new[] { source1, source2 });
+            comp.VerifyDiagnostics(
+                // (3,15): error CS0265: Partial declarations of 'C<T>' have inconsistent constraints for type parameter 'T'
```

Comment:
Probably unrelated, but this is a terrible error message. We should say _what_ the inconsistencies are.

Reply:
It looks like [https://github.com/dotnet/roslyn/issues/30229](https://github.com/dotnet/roslyn/issues/30229) is tracking this diagnostic.

---
In reply to: [273679942](https://github.com/dotnet/roslyn/pull/34850#discussion_r273679942) [](ancestors = 273679942)